src/System.Text.Json/tests/JsonDocumentTests.cs

```diff
@@ -919,11 +908,22 @@ public static void ReadTooPreciseDouble()
 
                 Assert.Equal(JsonValueType.Number, root.Type);
 
-                Assert.True(root.TryGetSingle(out float floatVal));
-                Assert.Equal(float.PositiveInfinity, floatVal);
+                if (PlatformDetection.IsFullFramework)
+                {
+                    Assert.False(root.TryGetSingle(out float floatVal));
```

Comment:
Should we be consistent with the reader and return false here/throw format exception?
https://github.com/dotnet/corefx/issues/33997#issuecomment-455724916

Reply:
> Should we be consistent with the reader

Yep.

> and return false here/throw format exception?

I'm not sure what you mean.  I think right now JsonDocument and Utf8JsonReader are consistent between themselves, in that it's `false`/throw on NetFx and `true`/infinity on CoreFx.

So we need to decide how we want them to behave:
* As-is
* NetFx behavior everywhere
* CoreFx behavior everywhere

The CoreFx behavior everywhere variant suggests we should delete the Try versions of GetSingle and GetDouble, since they'd be incapable of returning false.

Reply:
>  I think right now JsonDocument and Utf8JsonReader are consistent

Yes, you are right, but the issue I linked is proposing the change.

> The CoreFx behavior everywhere variant suggests we should delete the Try versions of GetSingle and GetDouble, since they'd be incapable of returning false.

Right. Given we have a source package that targets netstandard2.0, I think we should have NetFx behavior everywhere. Otherwise, the source package and the in-box for v1 will have different behaviors, which is less than ideal.

> we need to decide how we want them to behave

Yep, we should. I would like to here from @tannergooding and @KrzysztofCwalina, but I would prefer keeping the NetFx bevahior unless there is strong pushback.

Reply:
> I would like to here from @tannergooding and @KrzysztofCwalina, but I would prefer keeping the NetFx bevahior unless there is strong push-back.

I would push back and say that we should either have it "as-is" (CoreFX and NetFX differ) or that we should have CoreFX behavior everywhere. I believe the more consistent behavior is to have them differ as that is the experience users will already have with the majority of other APIs.

Having CoreFX return `true/infinity` makes it consistent with ES6 (which some people seem to care about) and keeps it IEEE compliant (which we've just spent a good amount of effort fixing, partially due to the number of bugs that have been filed in this general area).